### ISSUE TITLE
Feature request: Status Bar

### ISSUE THREAD

**Issue by hellocatfood** [synfig/synfig#1092]
Building on from the discussion on #1077 can Synfig have a Status Bar? This will display information about currently selected tools, layers and/or menu items such as what it does or how pressing different keys (Ctrl, Alt, Shift) will modify a tool. Especially useful for warnings 

![synfig](https://user-images.githubusercontent.com/151947/72671750-caa2a580-3a46-11ea-85cd-41eba40a0ce2.jpg)

There are several other design-focused open source programs which can be looked to for inspiration:

GIMP https://docs.gimp.org/2.10/en/gimp-image-window.html (GIMP's implementation is especially useful as the statusbar area blinks when there are errors or warnings)
Blender https://docs.blender.org/manual/en/latest/interface/window_system/status_bar.html
Inkscape http://tavmjong.free.fr/INKSCAPE/MANUAL/html/Anatomy.html#statusbar

**Comment by Jose-Moreno** [comment]
Hi. This sounds like a nice thing, currently however you can clearly see what you mocked in the status bar in dockers such as:

- The Tool Options, which holds descriptions for various tools. In fact in your screenshot / mockup you can clearly see this info on the right side of the screen. 
- The Info panel, where the cursor coordinates are located (as well as RGBA values and a render progress bar)
- The canvas view widget (where you see the zoom% and quick access buttons for adjusting the view scale) 

At first glance, these kind of seem to defeat the purpose of having a status bar, unless we argue that implementing such bar is meant to gather all those pieces of information in a single place for improved efficiency (and cleaning up wasted docker space). 

To that idea perhaps many of us can agree as it could be more convenient for a quick representation of important info, rather than scouring different widgets, this is particularly true when you're trying to use the dockers minimally when using a screen tablet for example.

In the case of zooming however I'd rather have the switchet on the canvas where it is now, as it's closer to your work area (and hopefully a % label can added to the navigator too :yum: ). A zoom label (read only) would be ok on the status bar though.

As for the **units switcher** you mocked up, those would be nice to have on the top bar near the rulers rather than on the status bar, so you can quickly see the change in units and not having to go up and down through the interface, however it may be argued that since those don't change as often  either, even between projects, they are fine to be left on the preferences (plus the interface updates must cost a lot in terms of performance :thinking: ).

What could be added to the status bar though is:
- Cursor coordinates
- RGBA info
- Render progress bar 
- Brief tool description
- Tool modes
- Tool shortcuts
- Zoom % (if it's only a label it should be fine, but changing it directly down there would be extremly inefficient)

At this point I see little use for the info panel if these options are migrated to a potential status bar. 

That said, some tools currently don't have all of their information or even descriptions exposed in the tools panel panel, so I think first it would be required that all tools expose this information (or we make new info) and then we can vote on where to put it along with the rest.

**Comment by hellocatfood** [comment]
You're right that my mockup doesn't make the best case for having a status bar. I basically took it from GIMP ;-) I didn't want to make a full proposal in teh first instance.

> Cursor coordinates
This might rely on being able to change the coordinates/ruler system and making it consistent #404 

> RGBA info
Of the pixel underneath the cursor? Would this not be better to displayed if the eyedropper tool is selected?

> Brief tool description
 For example, the tooltip for the eyedropper tells you what it is, but the Status Bar would tell you how to use it (Click on a image to get a colour)

> Tool modes
Hints at what keys can be pressed to change the mode, plus when the modifier key is pressed displayed updated information about what the tool now does

> Tool shortcuts
Currently these are displayed in tooltips. Should they be duplicated in the proposed Status Bar too?

Additionally, warnings when a tool is used incorrectly e.g. eyedropper doesn't appear to work on bitmaps, only vectors. Status Bar would warn of that.

> That said, some tools currently don't have all of their information or even descriptions exposed in the tools panel panel, so I think first it would be required that all tools expose this information
Do you know which tools don't have this information? We can work through them one by one.
